covsirphy/analysis/scenario.py

```diff
@@ -995,7 +999,115 @@ def score(self, metrics="RMSLE", variables=None, phases=None, past_days=None, na
+        records.fillna(method="ffill", inplace=True)
+
+        if len(records) < 1:
+            print("No records for this country")
+            return False
+
+        return records
+
+
+    def estimate_delay(self, oxcgrt_data, indicator="Stringency_index", 
+                        target="Infected", value_range=(7, 35)):
+        """
+
+        Estimate the average delay (in days) between a policy measure taken
+        and the moment of change in for the given target value, for example
+        `Infected`.The change over time is calculated with penalized change
+        point analysis. More information about the method of change point
+        detection can be found on:
+        https://ctruong.perso.math.cnrs.fr/ruptures-docs/build/html/detection/pelt.html.
+        Very long periods are taken out, using df.quantile(0.99).
+
+        Args:
+            oxcgrt_data (covsirphy.OxCGRTData): OxCGRT dataset
+            indicator (str): column variable from the OxCGRT dataset.
+            target (str): one of the target columns to detect changes in,
+                options are["Infected", "Confirmed", "Fatal", "Recovered"].
+                Uses self.records(). 
+            value_range (int, int): tuple, giving the minimum and maximum
+            range to search for change over time. It is assumed that when a
+            policy measure is taken, change will occur after at least 7 days
+            and before 35 days (=default).
+        Returns:
+            days: The estimated number of days of delay for a given country.
+            df: dataframe of target, indicator and calculated periods of change (in days).
+        """
+        records = self.ensure_records_confirmed(target)
+        records = records.set_index("Date")
+
+        oxcgrt_df = oxcgrt_data.subset(country=self.country).set_index("Date")
+        records[indicator] = oxcgrt_df[indicator]
+        records = records.reset_index()
+        df = records.pivot_table(index=target, values=indicator)
+        df_run = df.copy()
+        if len(df_run) < 1:
+            return self.delay_days, None
+
+        # Convert index to serial numbers
+        serial_df = pd.DataFrame(np.arange(1, df_run.index.max() + 1, 1))
+        serial_df.index += 1    
+   
+        df_run = df_run.join(serial_df, how="outer")
+        series = df_run.reset_index(drop=True).iloc[:, 0].dropna()
+
+        # Convert index values to Susceptible values
+        reset_series = series.reset_index(drop=True)
+        reset_series.index += 1
+        
+        # Detection with Ruptures
+        algorithm = rpt.Pelt(model="rbf", jump=1, min_size=value_range[0])
+
+        # Check for enough data for this country
+        try:
+            results = algorithm.fit_predict(series.values, pen=0.5)
+        except: 
+            print("Delay days could not be estimated for", self.country,", delay set to default: ", self.delay_days)
+            return self.delay_days, None
```

Comment:
Please return empty dataframe, not `None` because this will raise pandas errors when using the output.